### PR TITLE
Add FI permutation options to UnusedFeatures

### LINKED ISSUE #553: [FEAT] Add force permutation option in UnusedFeatures check
`UnusedFeatures` looks for the feature importance attr if possible, which is bias in some [sklearn models](https://explained.ai/rf-importance/).
I suggest adding an option to override this behaviour. This is already an option in `.utils.features.calculate_feature_importance`.

### PR DESCRIPTION
**Pull request by matanper**
resolve #553 

RN if we want to enable one of our global parameters for single check we have to edit the run. I'm open for suggestions of smarter ideas

### CODE REVIEW

File: deepchecks/checks/methodology/unused_features.py [hunk 1]
```diff
@@ -64,8 +65,26 @@ def __init__(self, feature_importance_threshold: float = 0.2, feature_variance_t
         self.n_top_unused_to_show = n_top_unused_to_show
         self.random_state = random_state
 
-    def run_logic(self, context: CheckRunContext) -> CheckResult:
-        """Run check.
+    def run(self,
+            train_dataset: Dataset,
+            test_dataset: Dataset,
+            model: BasicModel = None,
+            feature_importance_force_permutation: bool = None,
```
Comment (ItayGabbay) [inline deepchecks/checks/methodology/unused_features.py:72]:
```suggestion
            feature_importance_force_permutation: bool = False,
```

File: deepchecks/checks/methodology/unused_features.py [hunk 2]
```diff
@@ -64,8 +65,26 @@ def __init__(self, feature_importance_threshold: float = 0.2, feature_variance_t
...
+
+        Parameters
+        ----------
+        train_dataset : Dataset
+            dataset representing data an estimator was fitted on
+        test_dataset : Dataset
+            dataset representing data an estimator predicts on
+        model : BasicModel
+            A scikit-learn-compatible fitted estimator instance
+        feature_importance_force_permutation : bool , default None
```
Comment (ItayGabbay) [inline deepchecks/checks/methodology/unused_features.py:84]:
```suggestion
        feature_importance_force_permutation : bool , default: False
```

File: deepchecks/checks/methodology/unused_features.py [hunk 3]
```diff
@@ -64,8 +65,26 @@ def __init__(self, feature_importance_threshold: float = 0.2, feature_variance_t
...
+            force calculation of permutation features importance
+        feature_importance_timeout : int , default None
```
Comment (ItayGabbay) [inline deepchecks/checks/methodology/unused_features.py:86]:
```suggestion
        feature_importance_timeout : int , default: None
```

**Review by ItayGabbay — CHANGES_REQUESTED** [review]

File: deepchecks/checks/methodology/unused_features.py (hunk 3 above)
Comment (matanper) [reply deepchecks/checks/methodology/unused_features.py:86]: why? all the rest of docstrings doesn't have `:` after the default

File: deepchecks/checks/methodology/unused_features.py (hunk 1 above)
Comment (matanper) [reply deepchecks/checks/methodology/unused_features.py:72]: The idea of having it none is to have the default in a single place in the context init

File: deepchecks/checks/methodology/unused_features.py (hunk 3 above)
Comment (ItayGabbay) [reply deepchecks/checks/methodology/unused_features.py:86]: Where did you check?

Comment (matanper) [reply deepchecks/checks/methodology/unused_features.py:86]: I saw in Dataset, looks like there are both ways

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**